code/modules/mob/living/simple_animal/friendly/drone/_drone.dm

```diff
@@ -176,21 +176,21 @@
 	//Hands
 	for(var/obj/item/I in held_items)
 		if(!(I.flags_1 & ABSTRACT_1))
-			if(I.blood_DNA)
+			if(LAZYLEN(I.forensics.blood))
 				msg += "<span class='warning'>It has [icon2html(I, user)] [I.gender==PLURAL?"some":"a"] blood-stained [I.name] in its [get_held_index_name(get_held_index_of_item(I))]!</span>\n"
 			else
 				msg += "It has [icon2html(I, user)] \a [I] in its [get_held_index_name(get_held_index_of_item(I))].\n"
 
 	//Internal storage
-	if(internal_storage && !(internal_storage.flags_1&ABSTRACT_1))
-		if(internal_storage.blood_DNA)
+  if(internal_storage && !(internal_storage.flags_1&ABSTRACT_1))
```

Comment:
code/modules/mob/living/simple_animal/friendly/drone/_drone.dm:185:warning: if: if statement has no effect

I'm not sure exactly why it's warning on this line tbh, something about the flags is making it confused, it's possible it thinks you're trying to do (internal_storage && !internal_storage)

try
```
var/flagcheck = internal_storage.flags_1&ABSTRACT_1
if(internal_storage && flagcheck)
```